### PR TITLE
make awesome-slugify an optional requirement

### LINKED ISSUE #5: `awesome-slugify` dependency should be optional.
* Django Behaviors version: 0.4.0
* Django version: 2.0.5
* Python version: 3.6
* Operating System: Mac OS

### Description

I was previously using `django-behaviours==0.2.0`, but need to upgrade for Django 2 support. I have a dependency in my project, `unicode-slugify`, which clashes with `awesome-slugify` (both are imported via `import slugify`), but have incompatible APIs.

### What I Did

When I try and install version `0.4.0`, this clash breaks my project, as `awesome-slugify` is being imported in places where I mean to use `unicode-slugify`.

I think, as the `awesome-slugify` package is only required for the `Slugged` behaviour, the dependency should be listed in `extras`, rather than `install_requires`.

Happy to open a PR for this if you would like.

### PR DESCRIPTION
**Pull request by richardnias**
fixes #5.

By default, `awesome-slugify` will not be installed, unless you run `pip install django-behaviors[slugged]`.

If `awesome-slugify` is not present, we fall back to `django.utils.text.slugify`.

### CODE REVIEW

**Review by audiolion — APPROVED** [review]

**Comment by richardnias** [comment]
Finally got the test environment sorted out, sorry had been pulled on to other projects.

File: requirements_test.txt [hunk 1]
```diff
@@ -1,5 +1,3 @@
--r requirements.txt
```
Comment (audiolion) [inline requirements_test.txt:1]: why this change? to make the tox setup easier? I am fine with this, just  gotta edit the developer setup docs to be `pip install -r requirements.txt -r requirements_text.txt` if you want to run the tests locally

File: tox.ini [hunk 2]
```diff
@@ -1,8 +1,8 @@
 [tox]
 envlist =
-    {py27,py34,py35,py36}-django-18
-    {py27,py34,py35,py36}-django-111
-    {py34,py35,py36}-django-20
+    {py27,py34,py35,py36}-django-18-{slugify,noslugify}
```
Comment (audiolion) [inline tox.ini:3]: do you mind just explaining how this works for my own edification? I see that it is testing with awesome-slugify and without it but not sure how

File: requirements_test.txt (hunk 1 above)
Comment (richardnias) [reply requirements_test.txt:1]: Yeah — as the only thing in `requirements.txt` is `awesome-slugify`, which is now optional, it made sense to me to not include `requirements.txt` by default. 

an alternative would be to remove `awesome-slugify` from `requirements.txt` and explicitly include it in the deps in `tox.ini` — but then, `requirements.txt` will be an empty file and it wouldn't make sense to include it in `requirements_test.txt` anyway.

happy to change if you wish!

Comment (audiolion) [reply requirements_test.txt:1]: ah I see, no problem then all good

File: tox.ini (hunk 2 above)
Comment (richardnias) [reply tox.ini:3]: Sure! it was an educational exercise for myself too!

The line {py27,py34,py35,py36}-django-18-{slugify,noslugify}` creates 8 environments:
```
py27-django-18-slugify
py27-django-18-noslugify
...
py36-django-18-noslugify
```

line 15 (`slugify: -r{toxinidir}/requirements.txt`) tells tox that if the env contains 'slugify', it should install `awesome-slugify`.

Comment (richardnias) [reply tox.ini:3]: I realised after the fact that tox must actually be checking for the the presence of `slugify` not in the raw string name of the env, but in something like `envname.split('-')`, because otherwise _every_ env would install `awesome-slugify`, as "noslugify" obviously contains "slugify". However, I checked the output of tox and it seems that this has worked as intended.
